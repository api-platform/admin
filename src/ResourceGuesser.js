import React from 'react';
import { Resource } from 'react-admin';
import PropTypes from 'prop-types';
import ListGuesser from './ListGuesser';
import CreateGuesser from './CreateGuesser';
import EditGuesser from './EditGuesser';
import ShowGuesser from './ShowGuesser';

const ResourceGuesser = ({ list, edit, create, show, ...props }) => (
  <Resource
    {...props}
    create={create || CreateGuesser}
    edit={edit || EditGuesser}
    list={list || ListGuesser}
    show={show || ShowGuesser}
  />
);

ResourceGuesser.propTypes = {
  resource: PropTypes.string.isRequired,
};

export default ResourceGuesser;
