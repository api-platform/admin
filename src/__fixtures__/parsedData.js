import { Field } from '@api-platform/api-doc-parser';

export const ENTRYPOINT = 'http://entrypoint';

export const RESOURCE_NAME = 'user';

export const API_INPUT_DATA = {
  resources: [
    {
      name: `${RESOURCE_NAME}`,
      fields: [
        new Field('fieldA', {
          id: 'http://schema.org/fieldA',
          range: 'http://www.w3.org/2001/XMLSchema#string',
          reference: null,
          required: true,
        }),
        new Field('fieldB', {
          id: 'http://schema.org/fieldB',
          range: 'http://www.w3.org/2001/XMLSchema#string',
          reference: null,
          required: true,
        }),
        new Field('deprecatedField', {
          id: 'http://localhost/deprecatedField',
          range: 'http://www.w3.org/2001/XMLSchema#string',
          reference: null,
          required: true,
          deprecated: true,
        }),
      ],
      parameters: [],
      url: `${ENTRYPOINT}/users`,
    },
  ],
};

export const API_FIELDS_DATA = [
  new Field('fieldA', {
    id: 'http://schema.org/fieldA',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    required: true,
  }),
  new Field('fieldB', {
    id: 'http://schema.org/fieldB',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    required: true,
  }),
  new Field('deprecatedField', {
    id: 'http://localhost/deprecatedField',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    required: true,
    deprecated: true,
  }),
];
