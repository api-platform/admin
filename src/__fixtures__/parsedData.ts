// eslint-disable-next-line tree-shaking/no-side-effects-in-initialization
import { Api, Field, Parameter, Resource } from '@api-platform/api-doc-parser';

export const API_DATA = new Api('entrypoint', {
  resources: [
    new Resource('resource', '/resources', {
      fields: [new Field('bar')],
    }),
    new Resource('idSearchFilterResource', '/id_search_filter_resources', {
      parameters: [new Parameter('id', 'xmls:string', false, '')],
      getParameters: () => Promise.resolve([]),
    }),
  ],
});

const EmbeddedResource = new Resource('embedded', '/embeddeds', {
  fields: [new Field('address')],
});

export const API_FIELDS_DATA = [
  new Field('id', {
    id: 'http://schema.org/id',
    range: 'http://www.w3.org/2001/XMLSchema#integer',
    reference: null,
    embedded: null,
    required: false,
  }),
  new Field('fieldA', {
    id: 'http://schema.org/fieldA',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    required: true,
  }),
  new Field('fieldB', {
    id: 'http://schema.org/fieldB',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    required: true,
  }),
  new Field('deprecatedField', {
    id: 'http://localhost/deprecatedField',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    required: true,
    deprecated: true,
  }),
  new Field('title', {
    id: 'http://schema.org/title',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    required: false,
  }),
  new Field('description', {
    id: 'http://schema.org/description',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    required: false,
  }),
  new Field('nullText', {
    id: 'http://schema.org/nullText',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    required: false,
  }),
  new Field('embedded', {
    id: 'http://schema.org/embedded',
    reference: null,
    embedded: EmbeddedResource,
    maxCardinality: 1,
    required: false,
  }),
  new Field('embeddeds', {
    id: 'http://schema.org/embedded',
    reference: null,
    embedded: EmbeddedResource,
    required: false,
  }),
  new Field('formatType', {
    id: 'https://schema.org/BookFormatType',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    enum: {
      'Https://schema.org/ebook': 'https://schema.org/EBook',
      'Https://schema.org/audiobookformat':
        'https://schema.org/AudiobookFormat',
      'Https://schema.org/hardcover': 'https://schema.org/Hardcover',
    },
    required: false,
  }),
  new Field('status', {
    id: 'http://localhost/status',
    range: 'http://www.w3.org/2001/XMLSchema#string',
    reference: null,
    embedded: null,
    enum: { Available: 'AVAILABLE', 'Sold out': 'SOLD_OUT' },
    required: false,
  }),
];
