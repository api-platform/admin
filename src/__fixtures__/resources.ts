import type { Resource } from '@api-platform/api-doc-parser';

const resources: Resource[] = [
  {
    name: 'books',
    url: 'https://demo.api-platform.com/books',
    id: 'http://schema.org/Book',
    title: 'Book',
    fields: [
      {
        name: 'isbn',
        id: 'http://schema.org/isbn',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: false,
        description: 'The ISBN of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'title',
        id: 'http://schema.org/name',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'The title of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'description',
        id: 'http://schema.org/description',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'A description of the item',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'author',
        id: 'http://schema.org/author',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description:
          'The author of this content or rating. Please note that author is special in that HTML 5 provides a special mechanism for indicating authorship via the rel tag. That is equivalent to this and may be used interchangeably',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'publicationDate',
        id: 'http://schema.org/dateCreated',
        range: 'http://www.w3.org/2001/XMLSchema#dateTime',
        reference: null,
        required: true,
        description:
          'The date on which the CreativeWork was created or the item was added to a DataFeed',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'reviews',
        id: 'http://schema.org/reviews',
        range: 'http://schema.org/Review',
        reference: {
          name: 'reviews',
          url: 'https://demo.api-platform.com/reviews',
          id: 'http://schema.org/Review',
          title: 'Review',
          fields: [
            {
              name: 'body',
              id: 'http://schema.org/reviewBody',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description: 'The actual body of the review',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'rating',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/rating',
              range: 'http://www.w3.org/2001/XMLSchema#integer',
              reference: null,
              required: false,
              description: 'A rating',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'letter',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/letter',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description:
                'DEPRECATED (use rating now): A letter to rate the book',
              maxCardinality: null,
              deprecated: true,
            },
            {
              name: 'book',
              id: 'http://schema.org/itemReviewed',
              range: 'http://schema.org/Book',
              required: true,
              description: 'The item that is being reviewed/rated',
              maxCardinality: 1,
              deprecated: false,
            },
            {
              name: 'author',
              id: 'http://schema.org/author',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description: 'Author the author of the review',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'publicationDate',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/publicationDate',
              range: 'http://www.w3.org/2001/XMLSchema#dateTime',
              reference: null,
              required: false,
              description: 'Author the author of the review',
              maxCardinality: null,
              deprecated: false,
            },
          ],
          readableFields: [],
          writableFields: [],
          operations: [
            {
              name: 'Retrieves the collection of Review resources.',
              type: 'list',
              method: 'GET',
              returns: 'http://www.w3.org/ns/hydra/core#Collection',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/FindAction',
              ],
              deprecated: false,
            },
            {
              name: 'Creates a Review resource.',
              type: 'create',
              method: 'POST',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/CreateAction',
              ],
              deprecated: false,
            },
            {
              name: 'Retrieves Review resource.',
              type: 'show',
              method: 'GET',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/FindAction',
              ],
              deprecated: false,
            },
            {
              name: 'Deletes the Review resource.',
              type: 'delete',
              method: 'DELETE',
              returns: 'http://www.w3.org/2002/07/owl#Nothing',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/DeleteAction',
              ],
              deprecated: false,
            },
            {
              name: 'Replaces the Review resource.',
              type: 'edit',
              method: 'PUT',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/ReplaceAction',
              ],
              deprecated: false,
            },
            {
              name: 'Updates the Review resource.',
              type: 'edit',
              method: 'PATCH',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: ['http://www.w3.org/ns/hydra/core#Operation'],
              deprecated: false,
            },
          ],
          deprecated: false,
          parameters: [],
        },
        required: false,
        description: "The book's reviews",
        maxCardinality: null,
        deprecated: false,
      },
    ],
    readableFields: [
      {
        name: 'isbn',
        id: 'http://schema.org/isbn',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: false,
        description: 'The ISBN of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'title',
        id: 'http://schema.org/name',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'The title of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'description',
        id: 'http://schema.org/description',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'A description of the item',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'author',
        id: 'http://schema.org/author',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description:
          'The author of this content or rating. Please note that author is special in that HTML 5 provides a special mechanism for indicating authorship via the rel tag. That is equivalent to this and may be used interchangeably',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'publicationDate',
        id: 'http://schema.org/dateCreated',
        range: 'http://www.w3.org/2001/XMLSchema#dateTime',
        reference: null,
        required: true,
        description:
          'The date on which the CreativeWork was created or the item was added to a DataFeed',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'reviews',
        id: 'http://schema.org/reviews',
        range: 'http://schema.org/Review',
        reference: {
          name: 'reviews',
          url: 'https://demo.api-platform.com/reviews',
          id: 'http://schema.org/Review',
          title: 'Review',
          fields: [
            {
              name: 'body',
              id: 'http://schema.org/reviewBody',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description: 'The actual body of the review',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'rating',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/rating',
              range: 'http://www.w3.org/2001/XMLSchema#integer',
              reference: null,
              required: false,
              description: 'A rating',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'letter',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/letter',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description:
                'DEPRECATED (use rating now): A letter to rate the book',
              maxCardinality: null,
              deprecated: true,
            },
            {
              name: 'book',
              id: 'http://schema.org/itemReviewed',
              range: 'http://schema.org/Book',
              required: true,
              description: 'The item that is being reviewed/rated',
              maxCardinality: 1,
              deprecated: false,
            },
            {
              name: 'author',
              id: 'http://schema.org/author',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description: 'Author the author of the review',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'publicationDate',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/publicationDate',
              range: 'http://www.w3.org/2001/XMLSchema#dateTime',
              reference: null,
              required: false,
              description: 'Author the author of the review',
              maxCardinality: null,
              deprecated: false,
            },
          ],
          readableFields: [],
          writableFields: [],
          operations: [
            {
              name: 'Retrieves the collection of Review resources.',
              type: 'list',
              method: 'GET',
              returns: 'http://www.w3.org/ns/hydra/core#Collection',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/FindAction',
              ],
              deprecated: false,
            },
            {
              name: 'Creates a Review resource.',
              type: 'create',
              method: 'POST',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/CreateAction',
              ],
              deprecated: false,
            },
            {
              name: 'Retrieves Review resource.',
              type: 'show',
              method: 'GET',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/FindAction',
              ],
              deprecated: false,
            },
            {
              name: 'Deletes the Review resource.',
              type: 'delete',
              method: 'DELETE',
              returns: 'http://www.w3.org/2002/07/owl#Nothing',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/DeleteAction',
              ],
              deprecated: false,
            },
            {
              name: 'Replaces the Review resource.',
              type: 'edit',
              method: 'PUT',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/ReplaceAction',
              ],
              deprecated: false,
            },
            {
              name: 'Updates the Review resource.',
              type: 'edit',
              method: 'PATCH',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: ['http://www.w3.org/ns/hydra/core#Operation'],
              deprecated: false,
            },
          ],
          deprecated: false,
          parameters: [],
        },
        required: false,
        description: "The book's reviews",
        maxCardinality: null,
        deprecated: false,
      },
    ],
    writableFields: [
      {
        name: 'isbn',
        id: 'http://schema.org/isbn',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: false,
        description: 'The ISBN of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'title',
        id: 'http://schema.org/name',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'The title of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'description',
        id: 'http://schema.org/description',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'A description of the item',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'author',
        id: 'http://schema.org/author',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description:
          'The author of this content or rating. Please note that author is special in that HTML 5 provides a special mechanism for indicating authorship via the rel tag. That is equivalent to this and may be used interchangeably',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'publicationDate',
        id: 'http://schema.org/dateCreated',
        range: 'http://www.w3.org/2001/XMLSchema#dateTime',
        reference: null,
        required: true,
        description:
          'The date on which the CreativeWork was created or the item was added to a DataFeed',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'reviews',
        id: 'http://schema.org/reviews',
        range: 'http://schema.org/Review',
        reference: {
          name: 'reviews',
          url: 'https://demo.api-platform.com/reviews',
          id: 'http://schema.org/Review',
          title: 'Review',
          fields: [
            {
              name: 'body',
              id: 'http://schema.org/reviewBody',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description: 'The actual body of the review',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'rating',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/rating',
              range: 'http://www.w3.org/2001/XMLSchema#integer',
              reference: null,
              required: false,
              description: 'A rating',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'letter',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/letter',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description:
                'DEPRECATED (use rating now): A letter to rate the book',
              maxCardinality: null,
              deprecated: true,
            },
            {
              name: 'book',
              id: 'http://schema.org/itemReviewed',
              range: 'http://schema.org/Book',
              required: true,
              description: 'The item that is being reviewed/rated',
              maxCardinality: 1,
              deprecated: false,
            },
            {
              name: 'author',
              id: 'http://schema.org/author',
              range: 'http://www.w3.org/2001/XMLSchema#string',
              reference: null,
              required: false,
              description: 'Author the author of the review',
              maxCardinality: null,
              deprecated: false,
            },
            {
              name: 'publicationDate',
              id: 'https://demo.api-platform.com/docs.jsonld#Review/publicationDate',
              range: 'http://www.w3.org/2001/XMLSchema#dateTime',
              reference: null,
              required: false,
              description: 'Author the author of the review',
              maxCardinality: null,
              deprecated: false,
            },
          ],
          readableFields: [],
          writableFields: [],
          operations: [
            {
              name: 'Retrieves the collection of Review resources.',
              type: 'list',
              method: 'GET',
              returns: 'http://www.w3.org/ns/hydra/core#Collection',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/FindAction',
              ],
              deprecated: false,
            },
            {
              name: 'Creates a Review resource.',
              type: 'create',
              method: 'POST',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/CreateAction',
              ],
              deprecated: false,
            },
            {
              name: 'Retrieves Review resource.',
              type: 'show',
              method: 'GET',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/FindAction',
              ],
              deprecated: false,
            },
            {
              name: 'Deletes the Review resource.',
              type: 'delete',
              method: 'DELETE',
              returns: 'http://www.w3.org/2002/07/owl#Nothing',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/DeleteAction',
              ],
              deprecated: false,
            },
            {
              name: 'Replaces the Review resource.',
              type: 'edit',
              method: 'PUT',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: [
                'http://www.w3.org/ns/hydra/core#Operation',
                'http://schema.org/ReplaceAction',
              ],
              deprecated: false,
            },
            {
              name: 'Updates the Review resource.',
              type: 'edit',
              method: 'PATCH',
              expects: 'http://schema.org/Review',
              returns: 'http://schema.org/Review',
              types: ['http://www.w3.org/ns/hydra/core#Operation'],
              deprecated: false,
            },
          ],
          deprecated: false,
          parameters: [],
        },
        required: false,
        description: "The book's reviews",
        maxCardinality: null,
        deprecated: false,
      },
    ],
    operations: [
      {
        name: 'Retrieves the collection of Book resources.',
        type: 'list',
        method: 'GET',
        returns: 'http://www.w3.org/ns/hydra/core#Collection',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/FindAction',
        ],
        deprecated: false,
      },
      {
        name: 'Creates a Book resource.',
        type: 'create',
        method: 'POST',
        expects: 'http://schema.org/Book',
        returns: 'http://schema.org/Book',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/CreateAction',
        ],
        deprecated: false,
      },
      {
        name: 'Retrieves Book resource.',
        type: 'show',
        method: 'GET',
        returns: 'http://schema.org/Book',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/FindAction',
        ],
        deprecated: false,
      },
      {
        name: 'Replaces the Book resource.',
        type: 'edit',
        method: 'PUT',
        expects: 'http://schema.org/Book',
        returns: 'http://schema.org/Book',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/ReplaceAction',
        ],
        deprecated: false,
      },
      {
        name: 'Deletes the Book resource.',
        type: 'delete',
        method: 'DELETE',
        returns: 'http://www.w3.org/2002/07/owl#Nothing',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/DeleteAction',
        ],
        deprecated: false,
      },
    ],
    deprecated: false,
    parameters: [],
  },
  {
    name: 'parchments',
    url: 'https://demo.api-platform.com/parchments',
    id: 'https://demo.api-platform.com/docs.jsonld#Parchment',
    title: 'Parchment',
    fields: [
      {
        name: 'title',
        id: 'https://demo.api-platform.com/docs.jsonld#Parchment/title',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'The title of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'description',
        id: 'https://demo.api-platform.com/docs.jsonld#Parchment/description',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'A description of the item',
        maxCardinality: null,
        deprecated: false,
      },
    ],
    readableFields: [
      {
        name: 'title',
        id: 'https://demo.api-platform.com/docs.jsonld#Parchment/title',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'The title of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'description',
        id: 'https://demo.api-platform.com/docs.jsonld#Parchment/description',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'A description of the item',
        maxCardinality: null,
        deprecated: false,
      },
    ],
    writableFields: [
      {
        name: 'title',
        id: 'https://demo.api-platform.com/docs.jsonld#Parchment/title',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'The title of the book',
        maxCardinality: null,
        deprecated: false,
      },
      {
        name: 'description',
        id: 'https://demo.api-platform.com/docs.jsonld#Parchment/description',
        range: 'http://www.w3.org/2001/XMLSchema#string',
        reference: null,
        required: true,
        description: 'A description of the item',
        maxCardinality: null,
        deprecated: false,
      },
    ],
    operations: [
      {
        name: 'Retrieves the collection of Parchment resources.',
        type: 'list',
        method: 'GET',
        returns: 'http://www.w3.org/ns/hydra/core#Collection',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/FindAction',
        ],
        deprecated: true,
      },
      {
        name: 'Creates a Parchment resource.',
        type: 'create',
        method: 'POST',
        expects: 'https://demo.api-platform.com/docs.jsonld#Parchment',
        returns: 'https://demo.api-platform.com/docs.jsonld#Parchment',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/CreateAction',
        ],
        deprecated: true,
      },
      {
        name: 'Retrieves Parchment resource.',
        type: 'show',
        method: 'GET',
        returns: 'https://demo.api-platform.com/docs.jsonld#Parchment',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/FindAction',
        ],
        deprecated: true,
      },
      {
        name: 'Deletes the Parchment resource.',
        type: 'delete',
        method: 'DELETE',
        returns: 'http://www.w3.org/2002/07/owl#Nothing',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/DeleteAction',
        ],
        deprecated: true,
      },
      {
        name: 'Replaces the Parchment resource.',
        type: 'edit',
        method: 'PUT',
        expects: 'https://demo.api-platform.com/docs.jsonld#Parchment',
        returns: 'https://demo.api-platform.com/docs.jsonld#Parchment',
        types: [
          'http://www.w3.org/ns/hydra/core#Operation',
          'http://schema.org/ReplaceAction',
        ],
        deprecated: true,
      },
      {
        name: 'Updates the Parchment resource.',
        type: 'edit',
        method: 'PATCH',
        expects: 'https://demo.api-platform.com/docs.jsonld#Parchment',
        returns: 'https://demo.api-platform.com/docs.jsonld#Parchment',
        types: ['http://www.w3.org/ns/hydra/core#Operation'],
        deprecated: true,
      },
    ],
    deprecated: true,
    parameters: [],
  },
];

export default resources;
