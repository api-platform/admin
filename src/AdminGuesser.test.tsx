import React from 'react';
import ReactTestRenderer from 'react-test-renderer/shallow';
import { AdminResourcesGuesser } from './AdminGuesser.js';
import ResourceGuesser from './ResourceGuesser.js';
import resources from './__fixtures__/resources.js';
import { API_DATA } from './__fixtures__/parsedData.js';
import type {
  ApiPlatformAdminDataProvider,
  ApiPlatformAdminRecord,
} from './types.js';

const dataProvider: ApiPlatformAdminDataProvider = {
  getList: () => Promise.resolve({ data: [], total: 0 }),
  getOne: <RecordType extends ApiPlatformAdminRecord>() =>
    Promise.resolve({ data: { id: 'id' } } as { data: RecordType }),
  getMany: () => Promise.resolve({ data: [] }),
  getManyReference: () => Promise.resolve({ data: [], total: 0 }),
  update: <RecordType extends ApiPlatformAdminRecord>() =>
    Promise.resolve({ data: {} } as { data: RecordType }),
  updateMany: () => Promise.resolve({ data: [] }),
  create: <RecordType extends ApiPlatformAdminRecord>() =>
    Promise.resolve({ data: {} } as { data: RecordType }),
  delete: <RecordType extends ApiPlatformAdminRecord>() =>
    Promise.resolve({ data: { id: 'id' } } as { data: RecordType }),
  deleteMany: () => Promise.resolve({ data: [] }),
  introspect: () => Promise.resolve({ data: API_DATA }),
  subscribe: () => Promise.resolve({ data: null }),
  unsubscribe: () => Promise.resolve({ data: null }),
};

describe('<AdminGuesser />', () => {
  const renderer = ReactTestRenderer.createRenderer();

  test('renders loading', () => {
    renderer.render(
      <AdminResourcesGuesser
        resources={[]}
        loading
        includeDeprecated={false}
        dataProvider={dataProvider}
      />,
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('renders without custom resources', () => {
    renderer.render(
      <AdminResourcesGuesser
        resources={resources}
        loading={false}
        includeDeprecated={false}
        dataProvider={dataProvider}
      />,
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('renders with custom resources', () => {
    renderer.render(
      <AdminResourcesGuesser
        resources={resources}
        loading={false}
        includeDeprecated={false}
        dataProvider={dataProvider}>
        <ResourceGuesser name="custom" />
      </AdminResourcesGuesser>,
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('renders without data', () => {
    renderer.render(
      <AdminResourcesGuesser
        resources={[]}
        loading={false}
        includeDeprecated={false}
        dataProvider={dataProvider}
      />,
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
