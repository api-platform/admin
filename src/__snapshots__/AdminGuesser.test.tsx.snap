// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminGuesser /> renders loading 1`] = `
<Loading
  loadingPrimary="ra.page.loading"
  loadingSecondary="ra.message.loading"
/>
`;

exports[`<AdminGuesser /> renders with custom resources 1`] = `
<Admin
  dataProvider={
    {
      "create": [Function],
      "delete": [Function],
      "deleteMany": [Function],
      "getList": [Function],
      "getMany": [Function],
      "getManyReference": [Function],
      "getOne": [Function],
      "introspect": [Function],
      "subscribe": [Function],
      "unsubscribe": [Function],
      "update": [Function],
      "updateMany": [Function],
    }
  }
  i18nProvider={
    {
      "changeLocale": [Function],
      "getLocale": [Function],
      "getLocales": [Function],
      "translate": [Function],
    }
  }
  loading={[Function]}
  store={
    {
      "getItem": [Function],
      "removeItem": [Function],
      "removeItems": [Function],
      "reset": [Function],
      "setItem": [Function],
      "setup": [Function],
      "subscribe": [Function],
      "teardown": [Function],
    }
  }
>
  <ResourceGuesser
    name="custom"
  />
</Admin>
`;

exports[`<AdminGuesser /> renders without custom resources 1`] = `
<Admin
  dataProvider={
    {
      "create": [Function],
      "delete": [Function],
      "deleteMany": [Function],
      "getList": [Function],
      "getMany": [Function],
      "getManyReference": [Function],
      "getOne": [Function],
      "introspect": [Function],
      "subscribe": [Function],
      "unsubscribe": [Function],
      "update": [Function],
      "updateMany": [Function],
    }
  }
  i18nProvider={
    {
      "changeLocale": [Function],
      "getLocale": [Function],
      "getLocales": [Function],
      "translate": [Function],
    }
  }
  loading={[Function]}
  store={
    {
      "getItem": [Function],
      "removeItem": [Function],
      "removeItems": [Function],
      "reset": [Function],
      "setItem": [Function],
      "setup": [Function],
      "subscribe": [Function],
      "teardown": [Function],
    }
  }
>
  <ResourceGuesser
    name="books"
  />
</Admin>
`;

exports[`<AdminGuesser /> renders without data 1`] = `
<Admin
  dataProvider={
    {
      "create": [Function],
      "delete": [Function],
      "deleteMany": [Function],
      "getList": [Function],
      "getMany": [Function],
      "getManyReference": [Function],
      "getOne": [Function],
      "introspect": [Function],
      "subscribe": [Function],
      "unsubscribe": [Function],
      "update": [Function],
      "updateMany": [Function],
    }
  }
  i18nProvider={
    {
      "changeLocale": [Function],
      "getLocale": [Function],
      "getLocales": [Function],
      "translate": [Function],
    }
  }
  loading={[Function]}
  store={
    {
      "getItem": [Function],
      "removeItem": [Function],
      "removeItems": [Function],
      "reset": [Function],
      "setItem": [Function],
      "setup": [Function],
      "subscribe": [Function],
      "teardown": [Function],
    }
  }
/>
`;
